docs/redis.md

```diff
@@ -1,5 +1,94 @@
-# Redis queue API reference
+# Redis queue API
 
 Long-running deep learning models or batch processing is best architected as a queue. Cog has a built-in queue worker that can process predictions from a Redis queue, and return the output back to the queue.
 
-It's currently undocumented, but documentation is coming soon!
+The request queue is implemented with [Redis streams](https://redis.io/topics/streams-intro).
+
+## Start up the model
+
+The entrypoint to run a Cog model against a queue is `cog.server.redis_queue`. You need to provide the following positional arguments:
+
+- `redis_host`: the host your redis server is running on.
+- `redis_port`: the port your redis server is listening on.
+- `input_queue`: the queue the Cog model will listen to for prediction requests. This queue should already exist.
+- `upload_url`: the endpoint Cog will `PUT` output files to. <!-- link to file handling documentation -->
+- `consumer_id`: The name the Cog model will use to identify itself in the Redis group (also called "consumer name" by Redis).
+- `model_id`: a unique ID for the Cog model, used to label setup logs.
+- `log_queue`: the queue the Cog model should send setup logs to (prediction logs are sent as part of prediction responses).
+
+Note: logging is changing as part of 0.3.0, so the `model_id` and `log_queue` arguments are likely to change soon.
+
+You can optionally provide the following positional argument:
+
+- `predict_timeout`: the maximum time in seconds a prediction will be allowed to run for before it is terminated.
+
+For example:
+
+    docker run python -m cog.server.redis_queue \
```

Comment:
Not sure if commentary on the API/usage itself is out of scope for this PR since we're just dealing with docs here, but I'm curious why these are all positional arguments rather than [flags](https://clig.dev/#arguments-and-flags). Took me a bit of back and forth to understand what each argument represents is in this example.

Reply:
This is indeed weird. I'm guessing just tech debt. @andreasjansson do know why this is like this?

Reply:
[The code itself](https://github.com/replicate/cog/blob/main/python/cog/server/redis_queue.py#L309) acknowledges that "This is intensely fragile. This should be kwargs or JSON or something like that." Definitely worth rewriting this at some point.